go/libraries/doltcore/sqle/enginetest/dolt_harness.go

```diff
@@ -187,19 +196,32 @@ func (d *DoltHarness) NewDatabase(name string) sql.Database {
 	return d.NewDatabases(name)[0]
 }
 
-func (d *DoltHarness) NewDatabases(names ...string) []sql.Database {
+func (d *DoltHarness) initializeNewMultiRepoEnv() {
 	dEnv := dtestutils.CreateTestEnv()
-	d.env = dEnv
+	mrEnv, err := env.DoltEnvAsMultiEnv(context.Background(), dEnv)
+	require.NoError(d.t, err)
+	d.multiRepoEnv = mrEnv
+}
+
+func (d *DoltHarness) NewDatabases(names ...string) []sql.Database {
+	if d.multiRepoEnv == nil {
+		d.initializeNewMultiRepoEnv()
+	}
 
 	d.databases = nil
 	d.databaseGlobalStates = nil
 	for _, name := range names {
+		dEnv := dtestutils.CreateTestEnv()
```

Comment:
It makes me uncomfortable that all of the databases having the same working directory `/user/bheni/datasets/states`. This was a preexisting problem and doesn't appear to break anything right now, so maybe not a big deal.

Reply:
For example, `CreateTestEnv` could take a single in-memory filesystem and working directory for the new db. Again, maybe this is just a TODO.

Reply:
Yup, I was fairly surprised this worked actually. When I was first testing a fix I had the directory paths and the in-mem URL set up to be distinct for different envs. I tested without it and was able to take that out, but I'll take another look and see if there's a little bit more I can clean up here as part of this PR.

Reply:
I think isolating the working directories is a good call. I added a new form for `CreateTestEnv` and started using that within DoltHarness' `NewDatabase` function. This could definitely be pushed farther, but this felt like a good first step to help prevent test databases from interfering with each other in the future.